stdlib/ephemeron.ml

```diff
@@ -265,6 +267,49 @@ module GenHashTable = struct
       (* TODO inline 3 iterations *)
       find_rec_opt key hkey (h.data.(key_index h hkey))
 
+    let update h key f =
+      let rec loop h key (hkey: int) f = function
```

Comment:
Incidentally, this function (and `find_in_bucket` below) could be made TRMC. `find_in_bucket` in `hashtbl.ml` was made TRMC in #11354. But this can be left for a different PR.